odoo/tools/mail.py

```diff
@@ -549,6 +550,28 @@ def email_normalize(text):
         return False
     return emails[0].lower()
 
+
+def email_domain_extract(email):
+    """ Extract the company domain to be used by IAP services notably. Domain
+    is extracted from email information e.g:
+        - info@proximus.be -> proximus.be
+    """
+    normalized_email = email_normalize(email)
+    if normalized_email:
+        return normalized_email.split('@')[1]
+    return False
+
+def url_domain_extract(url):
+    """ Extract the company domain to be used by IAP services notably. Domain
+    is extracted from an URL e.g:
+        - www.info.proximus.be -> proximus.be
+    """
+    parser_results = urlparse(url)
+    company_hostname = parser_results.hostname
+    if company_hostname and '.' in company_hostname:
+        return '.'.join(company_hostname.split('.')[-2:])  # remove subdomains
```

Comment:
That won't work for URLs like `https://www.bbc.co.uk`, why isn't the hostname valid?

Reply:
Because it is a small heuristics to find proximus.be from www.info.proximus.be . Indeed .co.uk won't work. Such sadness.

Reply:
But why is it bad to enrich ~~a postiche~~ a domain that includes subdomain. For instance if the domain name is `coca.odoo.com` (small SME), you probably don't want to enrich with the same domain as `pepsi.odoo.com`. Maybe just remove the obvious `www` but keep the rest?
Also maybe the part checking on `localhost` or other fake domain should be in tools, not in `_get_company_domain`.

Reply:
Target is not pepsi.odoo.com. Target is that if you are Mr Proximus, your information is completed. I think for most people this will not bring much interest, as we target SMEs. Chassis Nivellois will probably not have much interest in auto-complete its company but hey, you never know.